Sources/Client/Extensions/Result+Extensions.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  Result+Extensions.swift
+//  StreamChatCore
+//
+//  Created by Alexey Bukhtin on 17/04/2019.
+//  Copyright © 2019 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+public extension Result {
+    /// Get the error from the result if it failed.
+    var error: Failure? {
+        if case .failure(let error) = self {
+            return error
+        }
+        
+        return nil
+    }
+    
+    /// Returns true if the result was success.
+    var isSuccess: Bool {
```

Comment:
Why not
```
var success: Success? {
  if case .success(let value) = self { return value }
  return nil
}
```
to mirror `var error` above?

Reply:
It could be sound like a success value, but it just checks if it was a success.
`error` is an error value.